api/envoy/extensions/filters/http/cache/v3/cache.proto

```diff
@@ -50,6 +51,48 @@ message CacheConfig {
     repeated config.route.v3.QueryParameterMatcher query_parameters_excluded = 4;
   }
 
+  message ThunderingHerdHandler {
+    // The default ``None`` handler will allow every request to go to the upstream service
+    // while a cache entry is not complete. This is the situation often described as
+    // "the thundering herd problem", in that many requests in a short time may overload
+    // the upstream server in a way one might expect a cache to prevent.
+    message None {
+    }
+
+    // The ``BlockUntilCompletion`` handler will allow one request to go to the upstream,
+    // then additional requests for that resource will be blocked until that initial one
+    // completes. After a request completes, all the blocked requests will be released to
+    // undergo the initial cache check again - if the first request succeeded and populated
+    // the cache, they will all read from the cache. If the first request failed, one
+    // request will be allowed to pass and the rest will be re-blocked.
+    message BlockUntilCompletion {
+      // If ``parallel_requests`` is set, then this many requests may be
+      // sent upstream simultaneously. This may address the problem where one slow or
+      // flaky client can potentially block all clients for the same resource, while still
+      // limiting the impact of thundering herd.
+      //
+      // If unset, defaults to 1.
+      uint32 parallel_requests = 1;
+
+      // If a positive ``unblock_additional_request_period`` is set, then while requests
+      // are blocked, if the given period expires, an additional request will be allowed
+      // through. This is to mitigate a problem potentially caused by the
+      // ``BlockUntilCompletion``, in which, because the filter writes to client and cache
+      // in parallel, a particularly slow client being selected as the "primary" can
+      // prevent the cache from becoming populated in a timely fashion. Allowing a trickle
+      // of additional requests improves the chance of the cache getting populated.
+      //
+      // If unset, blocked requests remain blocked until a previous request finishes.
+      google.protobuf.Duration unblock_additional_request_period = 2;
+    }
+
+    oneof handler {
+      None none = 1;
```

Comment:
Should this just be `google.protobug.Empty`? Or do you think there might be future content of the `None` message?

Reply:
There could potentially be future settings, and it also provides a consistent place for documentation of the behavior. :)

Reply:
Switching to google.protobuf.Empty though, because apparently at least one python protobuf compiler can't handle a message with no fields or annotations. Not really sure what this does to doccomments.